contracts/contracts/UserReplicaSetManager.sol

```diff
@@ -0,0 +1,428 @@
+        address _proposer2Address,
+        address _proposer3Address
+    );
+
+    /// @notice EIP-712 Typehash definitions
+    //          Used to validate identity with gasless transaction submission
+    bytes32 constant PROPOSE_ADD_UPDATE_CNODE_REQUEST_TYPEHASH = keccak256(
+        "ProposeAddOrUpdateContentNode(uint newCnodeId,address newCnodeDelegateOwnerWallet,uint proposerSpId,bytes32 nonce)"
+    );
+    bytes32 constant UPDATE_REPLICA_SET_REQUEST_TYPEHASH = keccak256(
+        "UpdateReplicaSet(uint userId,uint primary,bytes32 secondariesHash,uint oldPrimary,bytes32 oldSecondariesHash,bytes32 nonce)"
+    );
+
+    function initialize(
+        address _registryAddress,
+        bytes32 _userFactoryRegistryKey,
+        address _userReplicaSetBootstrapAddress,
+        uint _networkId
+    ) public initializer
+    {
+        require(
+            _registryAddress != address(0x00) &&
+            _userFactoryRegistryKey.length != 0,
+            "Requires non-zero _registryAddress and registryKey"
+        );
+        registry = RegistryInterface(_registryAddress);
+        userFactoryRegistryKey = _userFactoryRegistryKey;
+        userReplicaSetBootstrapAddress = _userReplicaSetBootstrapAddress;
+        // Initialize base Signing Logic contract
+        SigningLogicInitializable.initialize(
+            "User Replica Set Manager",
+            "1",
+            _networkId
+        );
+        seedComplete = false;
+    }
+
+    /**
+     * @notice Function used to initialize bootstrap nodes
+     * Prior to this function, the contract is effectively disabled
+     */
+    function seedBootstrapNodes(
+        uint[] calldata _bootstrapSPIds,
+        address[] calldata _bootstrapNodeDelegateWallets
+    ) external
+    {
+        require(
+            msg.sender == userReplicaSetBootstrapAddress,
+            "Only callable by userReplicaSetBootstrapAddress"
+        );
+        require(seedComplete == false, "Seed operation already completed");
+        uint256[3] memory emptyProposerIds = [uint256(0), uint256(0), uint256(0)];
+        require(
+            _bootstrapSPIds.length == _bootstrapNodeDelegateWallets.length,
+            "Mismatched bootstrap array lengths"
+        );
+        for (uint i = 0; i < _bootstrapSPIds.length; i++) {
+            spIdToContentNodeDelegateWallet[_bootstrapSPIds[i]] = _bootstrapNodeDelegateWallets[i];
+            emit AddOrUpdateContentNode(
+                _bootstrapSPIds[i],
+                _bootstrapNodeDelegateWallets[i],
+                emptyProposerIds,
+                address(0x00),
+                address(0x00),
+                address(0x00)
+            );
+        }
+        seedComplete = true;
+    }
+
+    /**
+     * @notice Chain of trust based authentication scheme
+     *         Nodes are required to have an identity in Audius L2 and this function enables
+     *         known entities to register other known entities on L2 contracts.
```

Comment:
some comments / documentation explaining who should be able to submit this request and why would be helpful, and how the chain of trust model works